player/lua/console.lua

```diff
@@ -107,6 +108,80 @@ mp.observe_property("user-data/osc/margins", "native", function(_, val)
     update()
 end)
 
+do
+    local width_length_ratio = 0.5
+    local osd_width, osd_height = 100, 100
+
+    ---Update osd resolution if valid
+    local function update_osd_resolution()
+        local dim = mp.get_property_native('osd-dimensions')
+        if not dim or dim.w <= 0 or dim.h <= 0 then
```

Comment:
I don't think these can be `< 0`, you could use `==`, though it's the same number of characters.

Reply:
I don't remember why I wrote it like that back in the day, maybe in case the implementation changes to (-1,-1) or something. It doesn't hurt to leave it as is, so I guess we can leave it?

Reply:
It doesn't matter either way, but I feel that returning negative dimensions would be a bug that we should notice and fix instead of hiding the error. Up to you though.